src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1962,4 +1973,20 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
         return this.queryRunner || this.connection.createQueryRunner("slave");
     }
 
+    /**
+     * Gets the selected column from the selection. Is not always the same as the selection itself:
+     * 'DISTINCT ON (foo.bar) foo.bar' has the columnPath of 'foo.bar'
+     */
+    protected getSelectionWithColumnPath(selection: string) {
+        let columnPath: string = selection;
+
+        if (/(distinct)/i.test(selection) && this.connection.driver instanceof PostgresDriver) {
```

Comment:
I would've put `this.connection.driver instanceof PostgresDriver`-condition before the regexp, but that way some tests broke with `this.connection` being undefined